core/network/impl/protocols/protocol_base_impl.hpp

```diff
@@ -84,8 +84,14 @@ namespace kagome::network {
       return context_->protocols_;
     }
 
-    std::optional<Protocol> const &protocol() const {
-      return context_->active_proto_;
+    std::optional<std::reference_wrapper<Protocol const>> protocol() const {
```

Comment:
Missed `active_proto_` from previous pr.

`active_proto_` doesn't work, because it's singleton (each protocol has single instance) and incoming streams may use both/all protocols (`active_proto_` will be overwritten).